cleanlab/datalab/adapter/imagelab.py

```diff
@@ -0,0 +1,196 @@
+"""Module for adapting Imagelab to the Datalab API.
+
+It uses the Adapter pattern so that Imagelab becomes compatible
+with the DataIssues, IssueFinder, Reporter classes.
+
+"""
+
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+from scipy.sparse import csr_matrix
+
+from cleanlab.datalab.data import Data
+from cleanlab.datalab.data_issues import DataIssues
+from cleanlab.datalab.issue_finder import IssueFinder
+from cleanlab.datalab.report import Reporter
+
+
+if TYPE_CHECKING:  # pragma: no cover
+    from datasets.arrow_dataset import Dataset
+    from cleanvision.imagelab import Imagelab
+
+
+def create_imagelab(dataset: "Dataset", image_key: Optional[str]) -> Optional["Imagelab"]:
+    """Creates Imagelab instance for running CleanVision checks. CleanVision checks are only supported for
+    huggingface datasets as of now.
+
+    Parameters
+    ----------
+    dataset: datasets.Dataset
+        Huggingface dataset used by Imagelab
+    image_key: str
+        key for image feature in the huggingface dataset
+
+    Returns
+    -------
+    Imagelab
+    """
+    imagelab = None
+    if not image_key:
+        return imagelab
+    try:
+        from cleanvision.imagelab import Imagelab
+        from datasets.arrow_dataset import Dataset
+
+        if isinstance(dataset, Dataset):
+            imagelab = Imagelab(hf_dataset=dataset, image_key=image_key, verbosity=0)
```

Comment:
what if the key does not exist in the dataset? will cleanvision output a helpful error message?

Reply:
make sure in this edge-case that the user does not get the "Cannot import datasets or cleanvision package..." error message

Reply:
Checked this edge case. It is handled in cleanvision where Imagelab throws error if key is not found and this error is propagated in datalab.